src/lint.cpp

```diff
@@ -356,6 +368,16 @@ void linter::visit_variable_use(identifier name) {
   this->visit_variable_use(name, used_variable_kind::use);
 }
 
+void linter::visit_variable_use_and_assignment(identifier name) {
+  QLJS_ASSERT(!this->scopes_.empty());
+  scope &current_scope = this->current_scope();
+  const declared_variable *var = current_scope.find_declared_variable(name);
+  if (!var) {
```

Comment:
Shouldn't we call `report_error_if_assignment_is_illegal` if `var` is not null? See `visit_variable_assignment`. I think your change causes the following code to no longer report an error:
```js
const x = 3;
x += 1;
```